progs/mcnode/main.go

```diff
@@ -2,30 +2,38 @@ package main
 
 import (
 	"context"
+	"encoding/json"
 	"errors"
 	"flag"
 	"fmt"
 	ggio "github.com/gogo/protobuf/io"
+	ggproto "github.com/gogo/protobuf/proto"
 	mux "github.com/gorilla/mux"
-	p2p_crypto "github.com/ipfs/go-libp2p-crypto"
 	p2p_peer "github.com/ipfs/go-libp2p-peer"
 	p2p_pstore "github.com/ipfs/go-libp2p-peerstore"
 	multiaddr "github.com/jbenet/go-multiaddr"
 	p2p_host "github.com/libp2p/go-libp2p/p2p/host"
 	p2p_net "github.com/libp2p/go-libp2p/p2p/net"
 	mc "github.com/mediachain/concat/mc"
 	pb "github.com/mediachain/concat/proto"
+	"io/ioutil"
 	"log"
 	"net/http"
 	"os"
+	"path"
+	"path/filepath"
+	"sync"
 	"time"
 )
 
 type Node struct {
-	id    p2p_peer.ID
-	privk p2p_crypto.PrivKey
-	host  p2p_host.Host
-	dir   p2p_pstore.PeerInfo
+	mc.Identity
+	host    p2p_host.Host
+	dir     p2p_pstore.PeerInfo
+	home    string
+	mx      sync.Mutex
+	stmt    map[string]*pb.Statement
```

Comment:
no, mc.Identity (which is embedded in the node struct) contains the private key and it's associated id.
So we have keys, but they are RSA for now and we are not signing until ECC keys are merged in libp2p.